File: drift/src/main/java/drift/com/drift/model/MessageRequestAttributes.java
package drift.com.drift.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.Date;

/**
 * Created by eoin on 28/02/2018.
 */

public class MessageRequestAttributes {

    /*
        "scheduleMeetingFlow": true,
        "scheduleMeetingWith":meetingUserId,
        "appointmentInfo":apointment]

     */
    @Expose
    @SerializedName("appointmentInfo")
    MessageRequestAttributesAppointment appointment;
    @Expose
    @SerializedName("scheduleMeetingWith")
    int meetingUserId;
    @Expose
    @SerializedName("scheduleMeetingFlow")
    boolean scheduleMeetingFlow = true;


    MessageRequestAttributes(GoogleMeeting googleMeeting, UserAvailability userAvailability, int meetingUserId, int conversationId, Date timeSlot) {

        this.meetingUserId = meetingUserId;
        this.appointment = new MessageRequestAttributesAppointment(googleMeeting, userAvailability, meetingUserId, conversationId, timeSlot);

    }

}

File: drift/src/main/java/drift/com/drift/model/MessageRequestAttributes.kt
package drift.com.drift.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

import java.util.Date

/**
 * Created by eoin on 28/02/2018.
 */

internal class MessageRequestAttributes (googleMeeting: GoogleMeeting, userAvailability: UserAvailability, meetingUserId: Int, conversationId: Int, timeSlot: Date) {

    /*
        "scheduleMeetingFlow": true,
        "scheduleMeetingWith":meetingUserId,
        "appointmentInfo":apointment]

     */

    @Expose
    @SerializedName("scheduleMeetingWith")
    internal var meetingUserId: Int = meetingUserId

    @Expose
    @SerializedName("appointmentInfo")
    internal var appointment: MessageRequestAttributesAppointment = MessageRequestAttributesAppointment(googleMeeting, userAvailability, meetingUserId, conversationId, timeSlot)
    @Expose
    @SerializedName("scheduleMeetingFlow")
    internal var scheduleMeetingFlow = true


}
